src/power_grid_model/_utils.py

```diff
@@ -439,11 +482,45 @@ def validate_data_filter(
 
 def is_sparse(component_data: ComponentData) -> bool:
     """Check if component_data is sparse or dense. Only batch data can be sparse."""
-    return isinstance(component_data, dict) and set(component_data.keys()) == {"indptr", "data"}
+    return isinstance(component_data, dict) and set(component_data.keys()) == {
+        "indptr",
+        "data",
+    }
 
 
 def is_columnar(component_data: ComponentData) -> bool:
     """Check if component_data is columnar or row based"""
     if is_sparse(component_data):
         return not isinstance(component_data["data"], np.ndarray)
     return not isinstance(component_data, np.ndarray)
+
+
+def validate_component_data(component_data: ComponentData, component=None) -> None:
```

Comment:
"validate" is a little bit ambiguous cause we kind of reserve that for the stuff in the data validator. it's probably a good idea to think about this properly

Reply:
Renamed to component_data_checks